contents/ruby/Rails.md

```diff
@@ -0,0 +1,154 @@
+
+For example, in order to fetch yesterday's date in Python, one may have to write
+
+```python
+import datetime
+datetime.datetime.now() - datetime.timedelta(days = 1)
+```
+
+The same action in Rails is done using
+```ruby
+1.day.ago
+```
+
+Similarly, Rails also added a new method to access an array element. Instead of following the C convention of using square-brackets to access the second element in the array "elements" like
+
+```c
+elements[1]
+```
+
+Rails developers can use ordinal representation to access the same element with the command
+
+```ruby
+elements.second
+```
+
+And while this may not reduce the amount one has to type, most Rails enthusiasts agree that it makes the code much more readable, intuitive, and inherently happiness-inducing.
+
+### Large and Active Community of Developers
+
+The Ruby language has a very passionate and extensive community of developers, and there is a vast array of libraries (called "Gems"), that can be used with your Rails projects. These libraries provide a lot of functionalities, and can be used to add flexibility to Rails projects when required. Furthermore, as a result of its popularity, it is very likely that any problems you encounter with Rails will have a solution that you can find on Google or StackOverflow. If not, the Ruby community is actually very famous for being 'nice', so you're likely to get a response to a new problem very quickly too. 
+
+However, Rails might not be best suited for all use cases, and there are certain areas where other frameworks might be more suitable. For instance:-
+
+### Drawbacks
+
+#### Lack of Flexibility
+
+As demonstrated above, Rails follows very strict conventions, that make life much easier for developers - or so Rails thinks. It might be true that there are some developers who find Rails' convention very stifling, and, as a result, prefer to use frameworks that allow them more flexibility. Rails will still allow you to do as you please, but since the effort of undo-ing the conventions makes the process longer and less enjoyable, it might be better to avoid Rails.
+
+#### Performance Time
+
+While newer releases have combatted this problem to a large degree, many of the old releases of Rails have had a reputation for being very bloated and slow to respond. Applications made using NodeJS, or purely client-side applications with a minimal backend, tend to have a much lower delay between an action and a response. While it should be noted that the vast majority of delays in applications occur due to poor optimization, and not the inherent speed of the framework, Rails still remains more bulky and slower than many of its competitors. 
```

Comment:
combatted -> combated

Reply:
> Applications made using NodeJS, or purely client-side applications with a minimal backend,

The phrasing seems to suggest that NodeJS specifically contrasts with client-side applications. I think you're using NodeJS as an example of some class of applications, but I'm not sure what exactly it is.

Reply:
Yeah I would prefer not to mention the names of other frameworks/backends explicitly. Just focus on how Rails has a lot of bulk and is written in Ruby which would run more slowly because it's an interpreted language.